internal/controller/policyrecommendation_registrar.go

```diff
@@ -99,13 +117,13 @@ func (r *PolicyRecommendationRegistrar) SetupWithManager(mgr ctrl.Manager) error
 		},
 	}
```

Comment:
From what I understand this is as good as processing all events since the other DeleteFunc. UpdateFunc, GenericFunc fields are not set the predicate functions of Delete(), Update(), Generic() are going to evaluate to true. This is as good as having no predicate specified. If the intention is to process only create events then other *Funcs need to be configured to return false.

Reply:
Good find. I assumed the default implementation would return false but it does return true which makes sense. Have added the impl of remaining functions.